core/include/gnuradio-4.0/ClaimStrategy.hpp

```diff
@@ -16,139 +16,164 @@
 
 namespace gr {
 
-struct NoCapacityException : public std::runtime_error {
-    NoCapacityException() : std::runtime_error("NoCapacityException"){};
-};
+namespace detail {
 
-// clang-format off
+/*
+ * `AtomicBitset` is a lock-free, thread-safe bitset.
+ * It allows for efficient and thread-safe manipulation of individual bits.
+ */
+template<std::size_t Size = std::dynamic_extent>
+class AtomicBitset {
```

Comment:
Should be covered by a dedicated unit-test.

Reply:
The unit test for `AtomicBitset` is in the `qa_buffer`.